src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,283 @@
+            foreach (var command in commands)
+            {
+                try
+                {
+                    Trace.Info($"Running powercfg.exe with {command}");
+                    using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                    {
+                        processInvoker.OutputDataReceived += delegate (object sender, ProcessDataReceivedEventArgs message)
+                        {
+                            Trace.Verbose(message.Data);
+                        };
+
+                        processInvoker.ErrorDataReceived += delegate (object sender, ProcessDataReceivedEventArgs message)
+                        {
+                            Trace.Error(message.Data);
+                        };
+
+                        processInvoker.ExecuteAsync(
+                                workingDirectory: string.Empty,
+                                fileName: filePath,
+                                arguments: command,
+                                environment: null,
+                                cancellationToken: CancellationToken.None);
+                    }
+                }
+                catch(Exception ex)
+                {
+                    //we will not stop the configuration. just show the warning and continue
+                    _terminal.WriteLine(StringUtil.Loc("PowerOptionsConfigError"));
+                    Trace.Error(ex);
+                }
+            }
+        }
+
+        private void AssertAdminAccess(bool unConfigure = false)
+        {
+            var windowsServiceHelper = HostContext.GetService<INativeWindowsServiceHelper>();
+            if (windowsServiceHelper.IsRunningInElevatedMode())
+            {
+                return;
+            }
+
+            if(unConfigure)
+            {
+                Trace.Error("Needs Administrator privileges to unconfigure an agent running with AutoLogon capability.");          
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonRemoval"));                
+            }
+            else
+            {
+                Trace.Error("Needs Administrator privileges to configure agent with AutoLogon capability.");
+                Trace.Error("You will need to unconfigure the agent and then re-configure with Administrative rights");            
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonCapability"));
+            }
+        }
+
+        //todo: move it to a utility class so that at other places it can be re-used
+        private void GetAccountSegments(string account, out string domain, out string user)
+        {
+            string[] segments = account.Split('\\');
+            domain = string.Empty;
+            user = account;
+            if (segments.Length == 2)
+            {
+                domain = segments[0];
+                user = segments[1];
+            }
+        }
+
+        private void WriteSection(string message)
```

Comment:
this is duplicated from ConfigurationManager. Windows service config doesn't WriteSection. It would be good to delete this here and just follow the same pattern.